com.unity.perception/Runtime/Randomization/Randomizers/SampleRandomizers/Utilities/PoissonDiskSampling.cs

```diff
@@ -0,0 +1,176 @@
+{
+    /// <summary>
+    /// Utility for generating lists of poisson disk sampled points
+    /// </summary>
+    public static class PoissonDiskSampling
+    {
+        const int k_DefaultSamplingResolution = 30;
+
+        // Algorithm sourced from Robert Bridson's paper "Fast Poisson Disk Sampling in Arbitrary Dimensions"
+        // https://www.cs.ubc.ca/~rbridson/docs/bridson-siggraph07-poissondisk.pdf
+        /// <summary>
+        /// Returns a list of poisson disc sampled points for a given area and density
+        /// </summary>
+        /// <param name="width">Width of the sampling area</param>
+        /// <param name="height">Height of the sampling area</param>
+        /// <param name="minimumRadius">The minimum distance required between each sampled point</param>
+        /// <param name="seed">The random seed used to initialize the algorithm state</param>
+        /// <param name="samplingResolution">The number of potential points sampled around every valid point</param>
+        /// <returns>The list of generated poisson points</returns>
+        public static NativeList<float2> GenerateSamples(
+            float width,
+            float height,
+            float minimumRadius,
+            uint seed = 12345,
+            int samplingResolution = k_DefaultSamplingResolution)
+        {
+            if (width < 0)
+                throw new ArgumentException($"Width {width} cannot be negative");
+            if (height < 0)
+                throw new ArgumentException($"Height {height} cannot be negative");
+            if (minimumRadius < 0)
+                throw new ArgumentException($"MinimumRadius {minimumRadius} cannot be negative");
+            if (seed == 0)
+                throw new ArgumentException("Random seed cannot be 0");
+            if (samplingResolution <= 0)
+                throw new ArgumentException($"SamplingAttempts {samplingResolution} cannot be <= 0");
+
+            var samples = new NativeList<float2>(Allocator.TempJob);
+            new SampleJob
+            {
+                width = width,
+                height = height,
+                minimumRadius = minimumRadius,
+                seed = seed,
+                samplingResolution = samplingResolution,
+                samples = samples
+            }.Schedule().Complete();
+            return samples;
+        }
+
+        [BurstCompile]
+        struct SampleJob : IJob
+        {
+            public float width;
+            public float height;
+            public float minimumRadius;
+            public uint seed;
+            public int samplingResolution;
+            public NativeList<float2> samples;
+
+            public void Execute()
+            {
+                var newSamples = Sample(width, height, minimumRadius, seed, samplingResolution, Allocator.Temp);
+                samples.AddRange(newSamples);
+                newSamples.Dispose();
+            }
+        }
+
+        static NativeList<float2> Sample(
```

Comment:
Just a general note: some comments in complicated math code explaining in regular english what is going on is helpful for not only code reviewers, but the poor software engineer (who may not be you) venturing into this in a later day to a) figure out what is going wrong, or b) trying to understand it to modify/enhance it

Reply:
I've added some comments to help explain the methodology behind this algorithm.